language/libra-vm/src/data_cache.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) The Libra Core Contributors
```

Comment:
this is effectively the Libra implementation of the data cache.
As the comment says it is also used by some random client for convenience and we may remove that later.
However I'd be fine to go out as is too given the only public usage is to create one.
It makes sense to me for this to live in the libra layer given it is exclusively a libra or client concept